src/PgManyToManyRelationPlugin.js

```diff
@@ -348,8 +853,132 @@ module.exports = function PgManyToManyRelationPlugin(
+          pgManyToManyJunctionLeftKeyAttributes: junctionLeftKeyAttributes,
+          pgManyToManyJunctionRightKeyAttributes: junctionRightKeyAttributes,
+        },
+        addArgDataGenerator,
+        Self,
+        field,
+      } = context;
+
+      if (!isPgManyToManyRelationField) return args;
+
+      const shouldAddCondition =
+        isPgFieldConnection || isPgFieldSimpleCollection;
+      if (!shouldAddCondition) return args;
+
+      const TableConditionType = getTypeByName(
+        inflection.conditionType(
+          inflection.manyToManyRelationByKeys(
+            leftKeyAttributes,
+            junctionLeftKeyAttributes,
+            junctionRightKeyAttributes,
+            rightKeyAttributes,
+            junctionTable,
+            rightTable,
+            junctionLeftConstraint,
+            junctionRightConstraint
+          )
+        )
+      );
+      if (!TableConditionType) {
+        return args;
+      }
+
+      const relevantAttributes = junctionTable.attributes.filter(
+        attr =>
+          pgColumnFilter(attr, build, context) &&
+          !omit(attr, "filter") &&
+          !junctionLeftKeyAttributes.includes(attr) &&
+          !junctionRightKeyAttributes.includes(attr)
+      );
+
+      addArgDataGenerator(function connectionCondition({ condition }) {
+        return {
+          pgQuery: queryBuilder => {
+            if (condition != null) {
+              // This is a bit precarious, and it doesn't yet work for live queries...
+              const junctionTableAlias = queryBuilder.junctionTableAlias;
+              if (!junctionTableAlias) {
+                throw new Error("Missing junctionTableAlias on queryBuilder");
+              }
+              relevantAttributes.forEach(attr => {
+                const fieldName = inflection.column(attr);
+                const val = condition[fieldName];
+                if (val != null) {
+                  // queryBuilder.addLiveCondition(() => record =>
+                  //   record[attr.name] === val
+                  // );
+                  queryBuilder.where(
+                    sql.fragment`${junctionTableAlias}.${sql.identifier(
+                      attr.name
+                    )} = ${gql2pg(val, attr.type, attr.typeModifier)}`
+                  );
+                } else if (val === null) {
+                  // queryBuilder.addLiveCondition(() => record =>
+                  //   record[attr.name] == null
+                  // );
+                  queryBuilder.where(
+                    sql.fragment`${junctionTableAlias}.${sql.identifier(
+                      attr.name
+                    )} IS NULL`
+                  );
+                }
+              });
+            }
+          },
+        };
+      });
+
+      // The `PgConnectionArgCondition` plugin adds a `condition` argument,
+      // but it doesn't do what we want. So we'll delete that argument,
+      // and set our own instead.
+      delete args.condition;
```

Comment:
Does it make sense to leave the `condition` argument intact, and do this kind of filtering using an `edgeCondition` argument? It's reasonable to allow developers to filter both on values on the junction table, _and_ on values on the right table. (I didn't consider that when I initially wrote this.)

Reply:
I’d rather condition wasn’t in the core API at all if I’m honest; people should only add the conditions that make sense. Rather than edgeConditions you want something like jointablename-conditions, I think. Or a conditions argument that takes conditions for both tables under separate (or combined?) fields.

Reply:
Fair enough. Filtering on values in the junction table is very important for my use case, so I definitely want to find a way to add that capability, but I do think it makes sense to do that in a separate plugin.

However, I also think that this repository has room for more than one plugin related to many-to-many relations. Maybe we could put that separate plugin in this repository, either in this pull request or in another one?